[BUG]: use `LazyFrame.collect_schema().names()` over `LazyFrame.columns`

**Describe the bug**
There is a perfromance warning in Polars 1.1.0 that using `data.columns` unecessarily collects the dataframe. It is better to use `LazyFrame.collect_schema().names()` to get names of the DF/LF.

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of pandera.
- [x] (optional) I have confirmed this bug exists on the main branch of pandera.

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example
The fix should be applied in `backend.polars.container/components/base`
```python
    def collect_column_info(self, check_obj: pl.LazyFrame, schema):
        """Collect column metadata for the dataframe."""
        column_names: List[Any] = []
        absent_column_names: List[Any] = []
        regex_match_patterns: List[Any] = []

        for col_name, col_schema in schema.columns.items():
            if (
                not col_schema.regex
                and col_name not in check_obj.columns
                and col_schema.required
            ):
                absent_column_names.append(col_name)

            if col_schema.regex:
                try:
                    column_names.extend(
                        col_schema.get_backend(check_obj).get_regex_columns(
                            col_schema, check_obj
                        )
                    )
                    regex_match_patterns.append(col_schema.selector)
                except SchemaError:
                    pass
            elif col_name in check_obj.columns:
                column_names.append(col_name)

        # drop adjacent duplicated column names
        destuttered_column_names = [*check_obj.columns]

        return ColumnInfo(
            sorted_column_names=dict.fromkeys(column_names),
            expanded_column_names=frozenset(column_names),
            destuttered_column_names=destuttered_column_names,
            absent_column_names=absent_column_names,
            regex_match_patterns=regex_match_patterns,
        )
```

#### Expected behavior
A clear and concise description of what you expected to happen.

#### Desktop (please complete the following information):

 - OS: macOS Sonoma
 - Browser: Safari
 - Version:
 -  pandera v 0.22.2
 - polars v 1.1.0
 - python 3.12.4

#### Screenshots
If applicable, add screenshots to help explain your problem.

#### Additional context
Add any other context about the problem here.

This makes `pytest` tests fail if it is using a Pandera.BaseModel for validation

```python
etf_data_samples = [<LazyFrame at 0x30C0D9BB0>, <LazyFrame at 0x30C27BD40>, <LazyFrame at 0x30C27BBC0>, <LazyFrame at 0x30C27B380>]

    @pytest.mark.asyncio()
    async def test_aget_asset_class_filter(etf_data_samples):
        """
        Test the aget_asset_class_filter function with various ETF data samples.

        This function also inherently tests the `normalize_asset_class` function.
        """
        for sample in etf_data_samples:
>           etf_data = ETFCategoryData(sample)

tests/unittests/portfolio/analytics/user_table/test_helpers.py:115:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
menv/lib/python3.12/site-packages/pandera/api/dataframe/model.py:138: in __new__
    DataFrameBase[TDataFrameModel], cls.validate(*args, **kwargs)
menv/lib/python3.12/site-packages/pandera/api/dataframe/model.py:289: in validate
    cls.to_schema().validate(
menv/lib/python3.12/site-packages/pandera/api/polars/container.py:58: in validate
    output = self.get_backend(check_obj).validate(
menv/lib/python3.12/site-packages/pandera/backends/polars/container.py:46: in validate
    column_info = self.collect_column_info(check_obj, schema)
menv/lib/python3.12/site-packages/pandera/backends/polars/container.py:214: in collect_column_info
    and col_name not in check_obj.columns
menv/lib/python3.12/site-packages/polars/lazyframe/frame.py:448: in columns
    issue_warning(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

message = 'Determining the column names of a LazyFrame requires resolving its schema, which is a potentially expensive operation. Use `LazyFrame.collect_schema().names()` to get the column names without this warning.'
category = <class 'polars.exceptions.PerformanceWarning'>, kwargs = {}

    def issue_warning(message: str, category: type[Warning], **kwargs: Any) -> None:
        """
        Issue a warning.

        Parameters
        ----------
        message
            The message associated with the warning.
        category
            The warning category.
        **kwargs
            Additional arguments for `warnings.warn`. Note that the `stacklevel` is
            determined automatically.
        """
>       warnings.warn(
            message=message, category=category, stacklevel=find_stacklevel(), **kwargs
        )
E       polars.exceptions.PerformanceWarning: Determining the column names of a LazyFrame requires resolving its schema, which is a potentially expensive operation. Use `LazyFrame.collect_schema().names()` to get the column names without this warning.

menv/lib/python3.12/site-packages/polars/_utils/various.py:444: PerformanceWarning
------------------- generated xml file: /Users/jjfantini/github/humblFINANCE-org/humblDATA/reports/pytest.xml -------------------
==================================================== short test summary info ====================================================
FAILED tests/unittests/portfolio/analytics/user_table/test_helpers.py::test_aget_asset_class_filter[lazyframe] - polars.exceptions.PerformanceWarning: Determining the column names of a LazyFrame requires resolving its schema, which is a potentially expensive operation. Use `LazyFrame.collect_schema().names()` to get the column names without this warning.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================ 1 failed, 16 passed, 3 skipped in 6.23s ============================================
```

--- a/.github/workflows/ci-tests.yml
+++ b/.github/workflows/ci-tests.yml
@@ -185,7 +185,7 @@ jobs:
         python-version: ["3.8", "3.9", "3.10", "3.11"]
         pandas-version: ["2.2.2"]
         pydantic-version: ["2.3.0"]
-        polars-version: ["0.20.31", "1.0.0"]
+        polars-version: ["0.20.31", "1.2.0"]
         extra:
         - hypotheses
         - io

--- a/.github/workflows/ci-tests.yml
+++ b/.github/workflows/ci-tests.yml
@@ -140,7 +140,7 @@ jobs:
       - name: Install deps
         run: |
           pip install uv
-          uv pip install -r requirements.in
+          uv pip install --system -r requirements.in
 
       - run: |
           pip list

--- a/.github/workflows/ci-tests.yml
+++ b/.github/workflows/ci-tests.yml
@@ -17,8 +17,6 @@ on:
 env:
   DEFAULT_PYTHON: 3.8
   CI: "true"
-  # Increase this value to reset cache if environment.yml has not changed
-  CACHE_VERSION: 6
 
 concurrency:
   group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
@@ -140,7 +138,7 @@ jobs:
             ${{ runner.os }}-pip-
 
       - name: Install deps
-        run:
+        run: |
           pip install uv
           uv pip install -r requirements.in
 

--- a/pandera/api/polars/utils.py
+++ b/pandera/api/polars/utils.py
@@ -1,13 +1,38 @@
+# pylint: disable=cyclic-import
 """Polars validation engine utilities."""
+
+from typing import Dict, List
 
 import polars as pl
 
 from pandera.api.polars.types import PolarsCheckObjects
+from pandera.engines.polars_engine import polars_version
 from pandera.config import (
     ValidationDepth,
     get_config_context,
     get_config_global,
 )
+
+
+def get_lazyframe_schema(lf: pl.LazyFrame) -> Dict[str, pl.DataType]:
+    """Get a dict of column names and  dtypes from a polars LazyFrame."""
+    if polars_version().release >= (1, 0, 0):
+        return lf.collect_schema()
+    return lf.schema
+
+
+def get_lazyframe_column_dtypes(lf: pl.LazyFrame) -> List[pl.DataType]:
+    """Get a list of column dtypes from a polars LazyFrame."""
+    if polars_version().release >= (1, 0, 0):
+        return lf.collect_schema().dtypes()
+    return [*lf.schema.values()]
+
+
+def get_lazyframe_column_names(lf: pl.LazyFrame) -> List[str]:
+    """Get a list of column names from a polars LazyFrame."""
+    if polars_version().release >= (1, 0, 0):
+        return lf.collect_schema().names()
+    return lf.columns
 
 
 def get_validation_depth(check_obj: PolarsCheckObjects) -> ValidationDepth:

--- a/pandera/backends/polars/base.py
+++ b/pandera/backends/polars/base.py
@@ -8,6 +8,7 @@ import polars as pl
 
 from pandera.api.base.error_handler import ErrorHandler
 from pandera.api.polars.types import CheckResult
+from pandera.api.polars.utils import get_lazyframe_column_dtypes
 from pandera.backends.base import BaseSchemaBackend, CoreCheckResult
 from pandera.constants import CHECK_OUTPUT_KEY
 from pandera.errors import (
@@ -21,8 +22,10 @@ from pandera.errors import (
 def is_float_dtype(check_obj: pl.LazyFrame, selector):
     """Check if a column/selector is a float."""
     return all(
-        dtype in pl.FLOAT_DTYPES
-        for dtype in check_obj.select(pl.col(selector)).schema.values()
+        dtype in {pl.Float32, pl.Float64}
+        for dtype in get_lazyframe_column_dtypes(
+            check_obj.select(pl.col(selector))
+        )
     )
 
 
@@ -155,9 +158,14 @@ class PolarsSchemaBackend(BaseSchemaBackend):
                 failure_cases_df = err.failure_cases
 
                 # get row number of the failure cases
-                index = err.check_output.with_row_count("index").filter(
-                    pl.col(CHECK_OUTPUT_KEY).eq(False)
-                )["index"]
+                if hasattr(err.check_output, "with_row_index"):
+                    _index_lf = err.check_output.with_row_index("index")
+                else:
+                    _index_lf = err.check_output.with_row_count("index")
+
+                index = _index_lf.filter(pl.col(CHECK_OUTPUT_KEY).eq(False))[
+                    "index"
+                ]
                 if len(err.failure_cases.columns) > 1:
                     # for boolean dataframe check results, reduce failure cases
                     # to a struct column

--- a/pandera/backends/polars/checks.py
+++ b/pandera/backends/polars/checks.py
@@ -10,6 +10,10 @@ from polars.lazyframe.group_by import LazyGroupBy
 from pandera.api.base.checks import CheckResult
 from pandera.api.checks import Check
 from pandera.api.polars.types import PolarsData
+from pandera.api.polars.utils import (
+    get_lazyframe_schema,
+    get_lazyframe_column_names,
+)
 from pandera.backends.base import BaseCheckBackend
 from pandera.constants import CHECK_OUTPUT_KEY
 
@@ -55,7 +59,7 @@ class PolarsCheckBackend(BaseCheckBackend):
         if isinstance(out, bool):
             return out
 
-        if len(out.columns) > 1:
+        if len(get_lazyframe_schema(out)) > 1:
             # for checks that return a boolean dataframe, reduce to a single
             # boolean column.
             out = out.select(
@@ -66,7 +70,11 @@ class PolarsCheckBackend(BaseCheckBackend):
                 ).alias(CHECK_OUTPUT_KEY)
             )
         else:
-            out = out.select(pl.col(out.columns[0]).alias(CHECK_OUTPUT_KEY))
+            out = out.select(
+                pl.col(get_lazyframe_column_names(out)[0]).alias(
+                    CHECK_OUTPUT_KEY
+                )
+            )
 
         return out
 
@@ -86,9 +94,9 @@ class PolarsCheckBackend(BaseCheckBackend):
         """Postprocesses the result of applying the check function."""
         results = pl.LazyFrame(check_output.collect())
         passed = results.select([pl.col(CHECK_OUTPUT_KEY).all()])
-        failure_cases = check_obj.lazyframe.with_context(results).filter(
-            pl.col(CHECK_OUTPUT_KEY).not_()
-        )
+        failure_cases = pl.concat(
+            [check_obj.lazyframe, results], how="horizontal"
+        ).filter(pl.col(CHECK_OUTPUT_KEY).not_())
 
         if check_obj.key is not None:
             failure_cases = failure_cases.select(check_obj.key)

--- a/pandera/backends/polars/components.py
+++ b/pandera/backends/polars/components.py
@@ -8,6 +8,10 @@ import polars as pl
 from pandera.api.base.error_handler import ErrorHandler
 from pandera.api.polars.components import Column
 from pandera.api.polars.types import PolarsData
+from pandera.api.polars.utils import (
+    get_lazyframe_schema,
+    get_lazyframe_column_names,
+)
 from pandera.backends.base import CoreCheckResult
 from pandera.backends.polars.base import PolarsSchemaBackend, is_float_dtype
 from pandera.config import ValidationDepth, ValidationScope, get_config_context
@@ -100,7 +104,7 @@ class ColumnBackend(PolarsSchemaBackend):
         return check_obj
 
     def get_regex_columns(self, schema, check_obj) -> Iterable:
-        return check_obj.select(pl.col(schema.selector)).columns
+        return get_lazyframe_schema(check_obj.select(pl.col(schema.selector)))
 
     def run_checks_and_handle_errors(
         self,
@@ -214,7 +218,7 @@ class ColumnBackend(PolarsSchemaBackend):
         isna = check_obj.select(expr)
         passed = isna.select([pl.col("*").all()]).collect()
         results = []
-        for column in isna.columns:
+        for column in get_lazyframe_column_names(isna):
             if passed.select(column).item():
                 continue
             failure_cases = (
@@ -326,8 +330,9 @@ class ColumnBackend(PolarsSchemaBackend):
 
         results = []
         check_obj_subset = check_obj.select(schema.selector)
-        for column in check_obj_subset.columns:
-            obj_dtype = check_obj_subset.schema[column]
+        for column, obj_dtype in get_lazyframe_schema(
+            check_obj_subset
+        ).items():
             results.append(
                 CoreCheckResult(
                     passed=schema.dtype.check(

--- a/pandera/backends/polars/container.py
+++ b/pandera/backends/polars/container.py
@@ -10,6 +10,7 @@ import polars as pl
 from pandera.api.base.error_handler import ErrorHandler
 from pandera.api.polars.container import DataFrameSchema
 from pandera.api.polars.types import PolarsData
+from pandera.api.polars.utils import get_lazyframe_column_names
 from pandera.backends.base import ColumnInfo, CoreCheckResult
 from pandera.backends.polars.base import PolarsSchemaBackend
 from pandera.config import ValidationDepth, ValidationScope, get_config_context
@@ -211,7 +212,7 @@ class DataFrameSchemaBackend(PolarsSchemaBackend):
         for col_name, col_schema in schema.columns.items():
             if (
                 not col_schema.regex
-                and col_name not in check_obj.columns
+                and col_name not in get_lazyframe_column_names(check_obj)
                 and col_schema.required
             ):
                 absent_column_names.append(col_name)
@@ -226,11 +227,11 @@ class DataFrameSchemaBackend(PolarsSchemaBackend):
                     regex_match_patterns.append(col_schema.selector)
                 except SchemaError:
                     pass
-            elif col_name in check_obj.columns:
+            elif col_name in get_lazyframe_column_names(check_obj):
                 column_names.append(col_name)
 
         # drop adjacent duplicated column names
-        destuttered_column_names = [*check_obj.columns]
+        destuttered_column_names = [*get_lazyframe_column_names(check_obj)]
 
         return ColumnInfo(
             sorted_column_names=dict.fromkeys(column_names),
@@ -256,7 +257,7 @@ class DataFrameSchemaBackend(PolarsSchemaBackend):
             from pandera.api.polars.components import Column
 
             columns = {}
-            for col in check_obj.columns:
+            for col in get_lazyframe_column_names(check_obj):
                 columns[col] = Column(schema.dtype, name=str(col))
 
         schema_components = []
@@ -579,7 +580,9 @@ class DataFrameSchemaBackend(PolarsSchemaBackend):
         )
 
         for lst in temp_unique:
-            subset = [x for x in lst if x in check_obj.columns]
+            subset = [
+                x for x in lst if x in get_lazyframe_column_names(check_obj)
+            ]
             duplicates = check_obj.select(subset).collect().is_duplicated()
             if duplicates.any():
                 failure_cases = check_obj.filter(duplicates)

--- a/pandera/engines/polars_engine.py
+++ b/pandera/engines/polars_engine.py
@@ -7,8 +7,10 @@ import inspect
 import warnings
 from typing import (
     Any,
+    Dict,
     Iterable,
     Literal,
+    Mapping,
     Optional,
     Sequence,
     Tuple,
@@ -19,7 +21,6 @@ from typing import (
 import polars as pl
 from packaging import version
 from polars.datatypes import DataTypeClass
-from polars.type_aliases import SchemaDict
 
 from pandera import dtypes, errors
 from pandera.api.polars.types import PolarsData
@@ -35,6 +36,9 @@ COERCION_ERRORS = (
     pl.exceptions.InvalidOperationError,
     pl.exceptions.ComputeError,
 )
+
+
+SchemaDict = Mapping[str, PolarsDataType]
 
 
 def polars_version() -> version.Version:
@@ -167,6 +171,8 @@ class DataType(dtypes.DataType):
         raises a :class:`~pandera.errors.ParserError` if the coercion fails
         :raises: :class:`~pandera.errors.ParserError`: if coercion fails
         """
+        from pandera.api.polars.utils import get_lazyframe_schema
+
         if isinstance(data_container, pl.LazyFrame):
             data_container = PolarsData(data_container)
 
@@ -187,7 +193,7 @@ class DataType(dtypes.DataType):
                 failure_cases = failure_cases.select(data_container.key)
             raise errors.ParserError(
                 f"Could not coerce {_key} LazyFrame with schema "
-                f"{data_container.lazyframe.schema} "
+                f"{get_lazyframe_schema(data_container.lazyframe)} "
                 f"into type {self.type}",
                 failure_cases=failure_cases,
                 parser_output=is_coercible,
@@ -561,16 +567,26 @@ class Array(DataType):
     def __init__(  # pylint:disable=super-init-not-called
         self,
         inner: Optional[PolarsDataType] = None,
+        shape: Union[int, Tuple[int, ...], None] = None,
+        *,
         width: Optional[int] = None,
     ) -> None:
-        if inner or width:
-            object.__setattr__(
-                self, "type", pl.Array(inner=inner, width=width)
-            )
+
+        kwargs: Dict[str, Union[int, Tuple[int, ...]]] = {}
+        if width is not None:
+            kwargs["shape"] = width
+        elif shape is not None:
+            kwargs["shape"] = shape
+
+        if inner or shape or width:
+            object.__setattr__(self, "type", pl.Array(inner=inner, **kwargs))
 
     @classmethod
     def from_parametrized_dtype(cls, polars_dtype: pl.Array):
-        return cls(inner=polars_dtype.inner, width=polars_dtype.width)
+        return cls(
+            inner=polars_dtype.inner,
+            shape=polars_dtype.shape,
+        )
 
 
 @Engine.register_dtype(equivalents=[pl.List])

--- a/tests/polars/test_polars_check.py
+++ b/tests/polars/test_polars_check.py
@@ -4,6 +4,7 @@ import polars as pl
 import pytest
 
 import pandera.polars as pa
+from pandera.api.polars.utils import get_lazyframe_schema
 from pandera.constants import CHECK_OUTPUT_KEY
 
 
@@ -145,7 +146,7 @@ def test_polars_element_wise_dataframe_check(lf):
     validated_data = schema.validate(lf)
     assert validated_data.collect().equals(lf.collect())
 
-    for col in lf.columns:
+    for col in get_lazyframe_schema(lf):
         invalid_lf = lf.with_columns(**{col: pl.Series([-1, 2, -4, 3])})
         try:
             schema.validate(invalid_lf)

--- a/tests/polars/test_polars_components.py
+++ b/tests/polars/test_polars_components.py
@@ -6,6 +6,7 @@ import polars as pl
 import pytest
 
 import pandera.polars as pa
+from pandera.api.polars.utils import get_lazyframe_schema
 from pandera.backends.base import CoreCheckResult
 from pandera.backends.polars.components import ColumnBackend
 from pandera.dtypes import DataType
@@ -83,7 +84,7 @@ def test_get_regex_columns(kwargs):
     backend = ColumnBackend()
     data = pl.DataFrame({f"col_{i}": [1, 2, 3] for i in range(10)}).lazy()
     matched_columns = backend.get_regex_columns(column_schema, data)
-    assert matched_columns == data.columns
+    assert matched_columns == get_lazyframe_schema(data)
 
     no_match_data = data.rename(
         lambda c: c.replace(
@@ -92,7 +93,7 @@ def test_get_regex_columns(kwargs):
         )
     )
     matched_columns = backend.get_regex_columns(column_schema, no_match_data)
-    assert matched_columns == []
+    assert len(matched_columns) == 0
 
 
 @pytest.mark.parametrize(

--- a/tests/polars/test_polars_config.py
+++ b/tests/polars/test_polars_config.py
@@ -6,6 +6,7 @@ import pytest
 
 import pandera.polars as pa
 from pandera.engines.polars_engine import polars_version
+from pandera.api.polars.utils import get_lazyframe_schema
 from pandera.api.base.error_handler import ErrorCategory
 from pandera.config import (
     CONFIG,
@@ -157,12 +158,12 @@ def test_coerce_validation_depth_none(validation_depth_none, schema):
     # simply calling validation shouldn't raise a coercion error, since we're
     # casting the types lazily
     validated_data = schema.validate(data)
-    assert validated_data.schema["a"] == pl.Int64
+    assert get_lazyframe_schema(validated_data)["a"] == pl.Int64
 
     ErrorCls = (
-        pl.InvalidOperationError
+        pl.exceptions.InvalidOperationError
         if polars_version().release >= (1, 0, 0)
-        else pl.ComputeError
+        else pl.exceptions.ComputeError
     )
     with pytest.raises(ErrorCls):
         validated_data.collect()

--- a/tests/polars/test_polars_container.py
+++ b/tests/polars/test_polars_container.py
@@ -13,6 +13,7 @@ from polars.testing.parametric import column, dataframes
 import pandera as pa
 from pandera import Check as C
 from pandera.api.polars.types import PolarsData
+from pandera.api.polars.utils import get_lazyframe_column_names
 from pandera.engines import polars_engine as pe
 from pandera.polars import Column, DataFrameModel, DataFrameSchema
 
@@ -383,14 +384,16 @@ def test_regex_selector(
 
         assert result.equals(ldf_for_regex_match.collect())
 
-        for column in ldf_for_regex_match.columns:
+        for column in get_lazyframe_column_names(ldf_for_regex_match):
             # this should raise an error since columns are not nullable by default
             modified_data = transform_fn(ldf_for_regex_match, column)
             with pytest.raises(pa.errors.SchemaError, match=exception_msg):
                 modified_data.pipe(schema.validate).collect()
 
         # dropping all columns should fail
-        modified_data = ldf_for_regex_match.drop(ldf_for_regex_match.columns)
+        modified_data = ldf_for_regex_match.drop(
+            get_lazyframe_column_names(ldf_for_regex_match)
+        )
         with pytest.raises(pa.errors.SchemaError):
             modified_data.pipe(schema.validate).collect()
 
@@ -549,7 +552,8 @@ def test_dataframe_schema_with_tz_agnostic_dates(time_zone, data):
     strategy = dataframes(
         column("datetime_col", dtype=pl.Datetime()),
         lazy=True,
-        size=10,
+        min_size=10,
+        max_size=10,
         allow_null=False,
     )
     lf = data.draw(strategy)

--- a/tests/polars/test_polars_dtypes.py
+++ b/tests/polars/test_polars_dtypes.py
@@ -14,6 +14,7 @@ from polars.testing.parametric import dataframes
 
 import pandera.errors
 from pandera.api.polars.types import PolarsData
+from pandera.api.polars.utils import get_lazyframe_column_dtypes
 from pandera.constants import CHECK_OUTPUT_KEY
 from pandera.engines import polars_engine as pe
 from pandera.engines.polars_engine import polars_object_coercible
@@ -29,6 +30,20 @@ def convert_object_to_decimal(
     return decimal.Decimal(number).quantize(
         decimal.Decimal(f"1e-{scale}"), decimal.ROUND_HALF_UP
     )
+
+
+POLARS_NUMERIC_DTYPES = [
+    pl.Int8,
+    pl.Int16,
+    pl.Int32,
+    pl.Int64,
+    pl.UInt8,
+    pl.UInt16,
+    pl.UInt32,
+    pl.UInt64,
+    pl.Float32,
+    pl.Float64,
+]
 
 
 numeric_dtypes = [
@@ -65,7 +80,12 @@ all_types = numeric_dtypes + temporal_types + other_types
 def get_dataframe_strategy(type_: pl.DataType) -> st.SearchStrategy:
     """Get a strategy for a polars dataframe of a given dtype."""
     return dataframes(
-        cols=2, lazy=True, allowed_dtypes=type_, null_probability=0.1, size=10
+        cols=2,
+        lazy=True,
+        allowed_dtypes=type_,
+        include_nulls=0.1,
+        min_size=10,
+        max_size=10,
     )
 
 
@@ -97,7 +117,7 @@ def test_coerce_no_cast(dtype, data):
 def test_coerce_no_cast_special(to_dtype, strategy):
     """Test that dtypes can be coerced without casting."""
     coerced = to_dtype.coerce(data_container=strategy)
-    for dtype in coerced.dtypes:
+    for dtype in get_lazyframe_column_dtypes(coerced):
         assert dtype == to_dtype.type
 
 
@@ -117,7 +137,7 @@ def test_coerce_cast(from_dtype, to_dtype, strategy, data):
     s = data.draw(strategy(from_dtype.type))
 
     coerced = to_dtype.coerce(data_container=s)
-    for dtype in coerced.dtypes:
+    for dtype in get_lazyframe_column_dtypes(coerced):
         assert dtype == to_dtype.type
 
 
@@ -138,7 +158,7 @@ def test_coerce_cast_special(pandera_dtype, data_container):
     """Test that dtypes can be coerced with casting."""
     coerced = pandera_dtype.coerce(data_container=data_container)
 
-    for dtype in coerced.dtypes:
+    for dtype in get_lazyframe_column_dtypes(coerced):
         assert dtype == pandera_dtype.type
 
     if isinstance(pandera_dtype, pe.Decimal):
@@ -152,9 +172,9 @@ def test_coerce_cast_special(pandera_dtype, data_container):
 
 
 ErrorCls = (
-    pl.InvalidOperationError
+    pl.exceptions.InvalidOperationError
     if pe.polars_version().release >= (1, 0, 0)
-    else pl.ComputeError
+    else pl.exceptions.ComputeError
 )
 
 
@@ -169,7 +189,7 @@ ErrorCls = (
         (
             pe.Bool(),
             pl.LazyFrame({"0": ["a", "b", "c"]}),
-            pl.InvalidOperationError,
+            pl.exceptions.InvalidOperationError,
         ),
         (
             pe.Int64(),
@@ -306,7 +326,7 @@ def test_polars_object_coercible(to_dtype, container, result):
     "inner_dtype_cls",
     [
         pl.Utf8,
-        *pl.NUMERIC_DTYPES,
+        *POLARS_NUMERIC_DTYPES,
     ],
 )
 @given(st.integers(min_value=2, max_value=10))
@@ -325,7 +345,7 @@ def test_polars_nested_array_type_check(inner_dtype_cls, width):
     "inner_dtype_cls",
     [
         pl.Utf8,
-        *pl.NUMERIC_DTYPES,
+        *POLARS_NUMERIC_DTYPES,
     ],
 )
 def test_polars_list_nested_type(inner_dtype_cls):
@@ -342,7 +362,7 @@ def test_polars_list_nested_type(inner_dtype_cls):
     "inner_dtype_cls",
     [
         pl.Utf8,
-        *pl.NUMERIC_DTYPES,
+        *POLARS_NUMERIC_DTYPES,
     ],
 )
 def test_polars_struct_nested_type(inner_dtype_cls):

--- a/tests/polars/test_polars_model.py
+++ b/tests/polars/test_polars_model.py
@@ -83,7 +83,7 @@ def ldf_model_with_custom_dataframe_checks():
         @dataframe_check
         @classmethod
         def not_empty(cls, data: PolarsData) -> pl.LazyFrame:
-            return data.lazyframe.select(pl.count().gt(0))
+            return data.lazyframe.select(pl.len().alias("len").gt(0))
 
     return ModelWithCustomDataFrameChecks
 
@@ -124,9 +124,9 @@ def test_model_schema_equivalency_with_optional():
 
 
 ErrorCls = (
-    pl.InvalidOperationError
+    pl.exceptions.InvalidOperationError
     if pe.polars_version().release >= (1, 0, 0)
-    else pl.ComputeError
+    else pl.exceptions.ComputeError
 )
 
 
@@ -243,7 +243,8 @@ def test_dataframe_schema_with_tz_agnostic_dates(time_zone, data):
     strategy = dataframes(
         column("datetime_col", dtype=pl.Datetime()),
         lazy=True,
-        size=10,
+        min_size=10,
+        max_size=10,
         allow_null=False,
     )
     lf = data.draw(strategy)

